pkg/txm/txstore.go

```diff
@@ -132,6 +133,30 @@ func (s *TxStore) GetTxState(lt uint64) (tracetracking.MsgStatus, bool, tvm.Exit
 	return tracetracking.NotFound, false, 0, tlb.ZeroCoins, false
 }
 
+// cleanupFinalizedAndExpired removes finalized transactions and expired unconfirmed transactions.
+// Returns the count of finalized and expired transactions that were removed. currentTimeMs is a Unix
+// timestamp in milliseconds.
+func (s *TxStore) cleanupFinalizedAndExpired(currentTimeMs uint64) (int, int) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	// Remove all finalized transactions
+	// TODO: consider selectively removing txs based on exit codes
+	finalizedCount := len(s.finalizedTxs)
+	s.finalizedTxs = map[uint64]*FinalizedTx{}
+
+	// Remove expired unconfirmed transactions
+	expiredCount := 0
+	for lt, unconfirmedTx := range s.unconfirmedTxs {
+		if unconfirmedTx.ExpirationMs < currentTimeMs {
```

Comment:
nit: I feel like it should be expired 'on' time
```suggestion
		if unconfirmedTx.ExpirationMs <= currentTimeMs {
```